expression/builtin_time_test.go

```diff
@@ -2155,7 +2155,7 @@ func (s *testEvaluatorSuite) TestMakeTime(c *C) {
 
 		{[]interface{}{0, 58.4, 0}, "00:58:00"},
 		{[]interface{}{0, "58.4", 0}, "00:58:00"},
-		{[]interface{}{0, 58.5, 1}, "00:59:01"},
+		{[]interface{}{0, 58.5, 1}, "00:58:01"},
```

Comment:
This test case can't be changed, because in MySQL:

```
MySQL [test]> select maketime(0, 58.5, 1);
+----------------------+
| maketime(0, 58.5, 1) |
+----------------------+
| 00:59:01             |
+----------------------+
1 row in set (0.000 sec)
```

I remembered that for time, it is rounded up, so you might want to use the original version for time.

Reply:
not exactly, I think in your case, 58.5 is treated as a decimal type, and in the unit test, it will be treated as type float. e.g. 
MySQL [test]> select maketime(0, 58.5, 1);
+----------------------+
| maketime(0, 585E-1, 1) |
+----------------------+
| 00:58:01             |
+----------------------+
1 row in set (0.000 sec)